lib/helper.spec.js

```diff
@@ -1439,63 +1439,207 @@ describe('findMaxRaise() ', () => {
-      last: new EWT(ABI_BET).bet(1, 100).sign(P2_KEY),
+      last: new EWT(ABI_BET).bet(1, 100).sign(P2_PRIV),
     }, {
       address: P3_ADDR,
-      last: new EWT(ABI_BET).bet(1, 100).sign(P3_KEY),
+      last: new EWT(ABI_BET).bet(1, 100).sign(P3_PRIV),
     }];
     expect(pokerHelper.findMinRaiseAmount(lineup, 2, 100)).to.equal(-1);
   });
 });
+
+
+describe('calcDistribution', () => {
+  it('should calculate distribution.', () => {
+    const lineup = [{
+      address: P_EMPTY,
+    }, {
+      address: P1_ADDR,
+      last: new EWT(ABI_FOLD).fold(2, 500).sign(P1_PRIV),
+    }, {
+      address: P2_ADDR,
+      last: new EWT(ABI_BET).bet(2, 1000).sign(P2_PRIV),
+    }];
+
+    expect(pokerHelper.calcDistribution(lineup, 'preflop', null, 10, P_EMPTY)).to.eql({
+      [P2_ADDR]: 1485,
+      [P_EMPTY]: 15,
+    });
+  });
+
+  it('should calc dist with empty seats.', () => {
+    const lineup = [{
+      address: P1_ADDR,
+      last: new EWT(ABI_FOLD).fold(2, 350).sign(P1_PRIV),
+    }, {
+      address: P2_ADDR,
+      last: new EWT(ABI_BET).bet(2, 850).sign(P2_PRIV),
+      sitout: 1,
+    }, {
+      address: P3_ADDR,
+      last: new EWT(ABI_SHOW).show(2, 850).sign(P3_PRIV),
+      cards: [9, 23],
+    }, {
+      address: P4_ADDR,
+      last: new EWT(ABI_FOLD).fold(2, 350).sign(P4_PRIV),
+    }, {
+      address: P_EMPTY,
+    }];
+    expect(pokerHelper.calcDistribution(lineup, 'showdown', null, 10, P_EMPTY)).to.eql({
+      [P3_ADDR]: 2376,
+      [P_EMPTY]: 24,
+    });
+  });
+
+  it('should calc dist for showdown with split pot and odd amounts.', () => {
+    const lineup = [{
+      address: P1_ADDR,
+      last: new EWT(ABI_SHOW).show(4, 1050).sign(P1_PRIV),
+      cards: [0, 1], // '2c', '3c'
+    }, {
+      address: P2_ADDR,
+      last: new EWT(ABI_SHOW).show(4, 1050).sign(P2_PRIV),
+      cards: [2, 3], // '4c', '5c'
+    }];
+    // board cards: 'Td', 'Jd', 'Qd', 'Kd', 'Ad'
+    expect(pokerHelper.calcDistribution(lineup, 'showdown', [21, 22, 23, 24, 25], 10, P_EMPTY)).to.eql({
+      [P1_ADDR]: 1039, // split pot1: 'Royal Flush'
+      [P2_ADDR]: 1039, // split pot1: 'Royal Flush'
+      [P_EMPTY]: 22,
+    });
+  });
+
+  it('should calc dist for sidepot.', () => {
+    const lineup = [{
+      address: P1_ADDR,
+      last: new EWT(ABI_SHOW).show(4, 1100).sign(P1_PRIV),
+      cards: [24, 25], // 'Kd', 'Ad'
+    }, {
+      address: P2_ADDR,
+      last: new EWT(ABI_SHOW).show(4, 400).sign(P2_PRIV),
+      sitout: 'allin',
+      cards: [0, 1],  // '2c', '3c'
+    }, {
+      address: P3_ADDR,
+      last: new EWT(ABI_SHOW).show(4, 1100).sign(P3_PRIV),
+      cards: [4, 5],  // '6c', '7c'
+    }, {
+      address: P4_ADDR,
+      last: new EWT(ABI_FOLD).fold(4, 50).sign(P4_PRIV),
+    }];
+    // board cards 'Td', 'Ac', '2d', '4c', '5c'
+    expect(pokerHelper.calcDistribution(lineup, 'showdown', [21, 12, 13, 2, 3], 10, P_EMPTY)).to.eql({
+      [P2_ADDR]: 1237, // pot1: 'Straight Flush, 5d High'
```

Comment:
@HelgeWieding why does poker solver say 5d High? there has never been a 5 of diamonds in the game.